index.html

```diff
@@ -86,25 +86,311 @@ <h3>Use-case #4: Avoiding "Hall of Mirrors"</h3>
+              constructor(CaptureActionEventInit init);
+              readonly attribute CaptureAction action;
+            };
+          </pre>
+          <dl data-link-for="CaptureActionEvent" data-dfn-for="CaptureActionEvent">
+            <dt>
+              <dfn>action</dfn>
+            </dt>
+            <dd>
+              The {{CaptureAction}} that was triggered.
+            </dd>
+          </dl>
+        </section>
+        <section>
+          <h4><dfn>CaptureActionEventInit</dfn></h4>
+          <pre class="idl">
+            dictionary CaptureActionEventInit : EventInit {
+              CaptureAction action;
+            };
+          </pre>
+          <dl
+            data-link-for="CaptureActionEventInit"
+            data-dfn-for="CaptureActionEventInit"
+          >
+            <dt>
+              <dfn>action</dfn>
+            </dt>
+            <dd>
+              The {{CaptureAction}} to initialize the event with.
+            </dd>
+          </dl>
+        </section>
+      </section>
+    </section>
+    <section id="capturing-side-actions">
+      <h2>Capturing Side for Actions</h2>
+      <p>
+        Capturing applications can enumerate available [=capture actions=] that
+        are supported on the video track they have obtained, by using
+        {{MediaStreamTrack/getSupportedCaptureActions}}, and can trigger those
+        actions by using {{MediaStreamTrack/sendCaptureAction}}.</li>
+      </ol>
+      <section id="capture-handle-get-supported-actions">
+        <h3>Enumerating supported actions and triggering them</h3>
+        <p>
+          When a {{MediaStreamTrack}} is a video track derived from screen-capture
+          of a <a data-cite="SCREEN-CAPTURE#dfn-browser">browser</a>
+          <a data-cite="SCREEN-CAPTURE#dfn-display-surface">display surface</a>,
+          {{MediaStreamTrack/getSupportedCaptureActions}} returns the set of
+          available [=capture actions=], if any, supported by the captured
+          application associated with this video track.
+        </p>
+        <pre class="idl">
+          partial interface MediaStreamTrack {
+            sequence&lt;DOMString&gt; getSupportedCaptureActions();
+            Promise&lt;undefined&gt; sendCaptureAction(CaptureAction action);
+          };
+        </pre>
+        <dl data-link-for="MediaStreamTrack" data-dfn-for="MediaStreamTrack">
+          <dt>
+            <dfn>getSupportedCaptureActions</dfn>
+          </dt>
+          <dd>
+            <p>When this method is invoked, the user agent MUST return [=this=]'
+            {{MediaDevices/[[AvailableCaptureActions]]}} if defined, or `[]` if not defined.</p>
+          </dd>
+          <dt>
+            <dfn>sendCaptureAction</dfn>
+          </dt>
+          <dd>
+            <p>When this method is invoked, the user agent MUST run the following steps:</p>
+            <ol>
+              <li>
+                If the [=relevant global object=] of [=this=] does not have
```

Comment:
Please see my other comments about how there is no guarantee that the user-activation has any correlation to the action being sent. Given that, I don't see the requirement of transient activation increases security. On the other hand, if we remove this requirement, we get closer to getting rid of the Promise, which I don't see as beneficial.

Reply:
Requiring transient activation seems reasonable given all intended use cases. There's no reason to give a capturing application more power than it needs.

> there is no guarantee that the user-activation has any correlation to the action being sent.

This is true of all transient activation.

Reply:
> There's no reason to give a capturing application more power than it needs.

1. What if the capturing application wants to send "jump 10 slides forward" or "fullscreen and unmute"? If we don't require+consume transient activation, such applications can just send multiple actions consecutively. This might be a bit janky, but it achieves the goal reasonably well without requiring the Web API to be changed. If we allowed that, I'd consider it a plus.
2. There's a risk that implementers of captured applications might place more trust in the source of an actions than they should, because they did not read our document thoroughly and carefully, and were lulled into a false sense of security.